include/zephyr/posix/pthread.h

```diff
@@ -153,7 +153,7 @@ int pthread_condattr_setclock(pthread_condattr_t *att, clockid_t clock_id);
  *
  * Initialize a rwlock with the default rwlock attributes.
  */
-#define PTHREAD_RWLOCK_INITIALIZER (-1)
+#define POSIX_RWLOCK_INITIALIZER (-1)
```

Comment:
hey @cfriedt is this correct? I bumped into a library that was using `PTHREAD_RWLOCK_INITIALIZER` (boringssl) and now does not build anymore, I see that `/usr/include/pthread.h` on my system defines `PTHREAD_RWLOCK_INITIALIZER` as well

Reply:
@fabiobaltieri - good find. `PTHREAD_RWLOCK_INITIALIZER` is the standard macro name. Are you good to submit a fix?

Reply:
yeah I can do that

Reply:
there you go https://github.com/zephyrproject-rtos/zephyr/pull/73808